### PR TITLE
[#5927] Add checksum verification keywords to replication API and microservice (4-2-stable)

### PR DESCRIPTION
**Pull request by alanking**
I split this up into two parts: the API changes and the microservice changes. Willing to squash it down.

This passed in CI

### CODE REVIEW

**Comment by alanking** [comment]
Fixed unused import Codacy issue, but use of `strncpy` is fine, so ignoring that one.

File: scripts/irods/test/test_all_rules.py [hunk 1]
```diff
@@ -1294,3 +1294,228 @@ def test_msiGetStderrInExecCmdOut_does_not_segfault_when_using_failed_out_parame
...
+        super(Test_msiDataObjRepl_checksum_keywords, self).tearDown()
+
+
+    def get_rule_text(self, _keywords=None):
+        from textwrap import dedent
+
+        keywords = '++++'.join(['destRescName={}'.format(self.target_resc)] +
+                               (_keywords or list()))
+
+        # TODO: add to rule texts or come up with some better way to do this?
```
Comment (trel) [inline scripts/irods/test/test_all_rules.py:1326]: either here or rule texts is fine.

Comment (alanking) [reply scripts/irods/test/test_all_rules.py:1326]: Okay. I'll just leave it as-is for now.

**Comment by alanking** [comment]
Fixed commit message and squashed

**Comment by trel** [comment]
looking good.   please # when ready/green/tested/happy.

**Comment by alanking** [comment]
#'d

**Pull request merged**